language/move-binary-format/src/control_flow_graph.rs

```diff
@@ -46,12 +50,27 @@ struct BasicBlock {
 pub struct VMControlFlowGraph {
     /// The basic blocks
     blocks: Map<BlockId, BasicBlock>,
+    /// Basic block ordering for traversal
+    traversal_successors: Map<BlockId, BlockId>,
+    /// Map from blocks with a loop's last continue to that loop's head
+    /// "last" here meaning at the highest bytecode offset
+    /// "continue" simply being a back-edge, but it is a continue in the sense that the loop
+    /// back-edge resembles a "continue" in its control flow layout
+    loop_last_continue_blocks: Map<BlockId, BlockId>,
```

Comment:
What is the difference between a "loop last continue" and a back edge? It sounds like all loop last continues are back edges, but not the other way around. Are there other differences? Basically, I'm trying to figure out why we use these and not back edges...

Reply:
It's the last back edge. But maybe you are right and you can just use the first back edge? In some ways, I think this might be an over optimization?

A loop here is a series of blocks. There might be many back edges in that set of blocks to the loop head. With this there are two options:
- Revisit the loop in the absint traversal when any backedge changes the pre state of the loop head's block 
- Revisit the loop in the absint traversal once all blocks with the loop head's block as a successor are visited

Setting up this "last continue" nonsense takes the latter approach. You only revisit the loop head block once the entirety of the loop has been visited at least once